src/modules/auth/login.strategy.ts

```diff
@@ -2,29 +2,41 @@ import { LoginStrategy } from 'passport-did-auth';
 import { Inject, Injectable } from '@nestjs/common';
 import { PassportStrategy } from '@nestjs/passport';
 import { ConfigService } from '@nestjs/config';
+import { verifyCredential } from 'didkit-wasm-node';
+import { RoleIssuerResolver } from '../claim/resolvers/issuer.resolver';
+import { RoleRevokerResolver } from '../claim/resolvers/revoker.resolver';
+import { RoleCredentialResolver } from '../claim/resolvers/credential.resolver';
 
 @Injectable()
 export class AuthStrategy extends PassportStrategy(LoginStrategy, 'login') {
   constructor(
     configService: ConfigService,
-    @Inject('IPFSClientConfig') ipfsConfig
+    @Inject('IPFSClientConfig') ipfsConfig,
+    issuerResolver: RoleIssuerResolver,
+    revokerResolver: RoleRevokerResolver,
+    credentialResolver: RoleCredentialResolver
   ) {
-    let loginStrategyParams = {
-      name: 'login',
-      rpcUrl: configService.get<string>('ENS_URL'),
-      cacheServerUrl: configService.get<string>('STRATEGY_CACHE_SERVER'),
-      privateKey: configService.get<string>('STRATEGY_PRIVATE_KEY'),
-      didContractAddress: configService.get<string>('DID_REGISTRY_ADDRESS'),
-      ensRegistryAddress: configService.get<string>('ENS_REGISTRY_ADDRESS'),
-      ipfsUrl: ipfsConfig,
-    };
-    const numBlocksBack = configService.get<string>('STRATEGY_NUM_BLOCKS_BACK');
-    if (numBlocksBack) {
-      loginStrategyParams = {
-        ...loginStrategyParams,
-        ...{ numberOfBlocksBack: numBlocksBack },
-      };
+    const loginStrategyParams: ConstructorParameters<typeof LoginStrategy> = [
+      {
+        name: 'login',
+        rpcUrl: configService.get<string>('ENS_URL'),
+        cacheServerUrl: configService.get<string>('STRATEGY_CACHE_SERVER'),
+        privateKey: configService.get<string>('STRATEGY_PRIVATE_KEY'),
+        didContractAddress: configService.get<string>('DID_REGISTRY_ADDRESS'),
+        ensRegistryAddress: configService.get<string>('ENS_REGISTRY_ADDRESS'),
+        ipfsUrl: ipfsConfig,
+      },
+      issuerResolver,
+      revokerResolver,
+      credentialResolver,
+      verifyCredential,
+    ];
+    const numberOfBlocksBack = parseInt(
+      configService.get<string>('STRATEGY_NUM_BLOCKS_BACK')
+    );
+    if (numberOfBlocksBack) {
+      loginStrategyParams[0].numberOfBlocksBack = numberOfBlocksBack;
```

Comment:
I think it's slightly nicer to keep a reference to `loginStrategyOptions` so that `numberOfBlocksBack` can be added without using `0` index. But not a big deal.
```suggestion
    const loginStrategyOptions: LoginStrategyOptions = {
        name: 'login',
        rpcUrl: configService.get<string>('ENS_URL'),
        cacheServerUrl: configService.get<string>('STRATEGY_CACHE_SERVER'),
        privateKey: configService.get<string>('STRATEGY_PRIVATE_KEY'),
        didContractAddress: configService.get<string>('DID_REGISTRY_ADDRESS'),
        ensRegistryAddress: configService.get<string>('ENS_REGISTRY_ADDRESS'),
        ipfsUrl: ipfsConfig,
     }
    const loginStrategyParams: ConstructorParameters<typeof LoginStrategy> = [
      loginStrategyOptions,
      issuerResolver,
      revokerResolver,
      credentialResolver,
      verifyCredential,
    ];
    const numberOfBlocksBack = parseInt(
      configService.get<string>('STRATEGY_NUM_BLOCKS_BACK')
    );
    if (numberOfBlocksBack) {
      loginStrategyOptions.numberOfBlocksBack = numberOfBlocksBack;
```

It should work, I did a quick test in node repl:
```
> const nested = {}
undefined
> const outer = [nested]
undefined
> nested.prop = 1
1
> outer
[ { prop: 1 } ]
```

Reply:
yes, it allowes accidentialy assign `numberOfBlocksBack` to some other property of loginStrategyOptions